azurerm/resource_arm_virtual_machine_scale_set.go

```diff
@@ -393,24 +394,34 @@ func resourceArmVirtualMachineScaleSet() *schema.Resource {
 				MaxItems: 1,
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
```

Comment:
you are right @tombuildsstuff the update operation should not force new VMSS. I have updated my code to reflect that.